### PR TITLE
[SPARK-21895] [SQL] Support changing database in HiveClient

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
Supporting moving tables across different database in HiveClient `alterTable`

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @cloud-fan @jiangxb1987

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 1]
```diff
@@ -495,15 +495,18 @@ private[hive] class HiveClientImpl(
...
+      table: CatalogTable): Unit = withHiveState {
     // getTableOption removes all the Hive-specific properties. Here, we fill them back to ensure
     // these properties are still available to the others that share the same Hive metastore.
     // If users explicitly alter these Hive-specific properties through ALTER TABLE DDL, we respect
     // these user-specified values.
     val hiveTable = toHiveTable(
       table.copy(properties = table.ignoredProperties ++ table.properties), Some(userName))
     // Do not use `table.qualifiedName` here because this may be a rename
-    val qualifiedTableName = s"${table.database}.$tableName"
+    val qualifiedTableName = s"$dbName.$tableName"
```
Comment (jiangxb1987) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: Should we add comment to explain this change?

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM, too.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala (hunk 1 above)
Comment (gatorsmile) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: This change does not change the semantics.

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**